File: app/src/main/java/com/ishow/noah/modules/account/modify/ModifyUserInfoContract.java
package com.ishow.noah.modules.account.modify;

import android.content.Context;

import com.ishow.common.mvp.base.IPresenter;
import com.ishow.common.mvp.base.IView;
import com.ishow.common.mvp.base.IViewStatus;
import com.ishow.noah.entries.User;

interface ModifyUserInfoContract{

    interface View extends IView, IViewStatus {

        /**
         * 更新成功
         */
        void updateAvatar(String avatar);
    }

    interface Presenter extends IPresenter {
        /**
         * 修改头像
         */
        void modifyAvatar(Context context, String avatar);

        /**
         * 更新用户信息
         */
        void modifyUserInfo(Context context, User user);
    }
}

File: app/src/main/java/com/ishow/noah/modules/account/modify/ModifyUserInfoContract.kt
package com.ishow.noah.modules.account.modify

import android.content.Context

import com.ishow.common.app.mvp.IPresenter
import com.ishow.common.app.mvp.IView
import com.ishow.noah.entries.User

/**
 * Created by yuhaiyang on 2018/8/8.
 * 修改用户信息-Contract
 */
internal interface ModifyUserInfoContract {

    interface View : IView {

        /**
         * 更新成功
         */
        fun updateAvatar(avatar: String)
    }

    interface Presenter : IPresenter {
        /**
         * 修改头像
         */
        fun modifyAvatar(context: Context, avatar: String)

        /**
         * 更新用户信息
         */
        fun modifyUserInfo(context: Context, user: User)
    }
}